python/pyspark/sql/tests/test_functions.py

```diff
@@ -286,6 +289,42 @@ def test_dayofweek(self):
         row = df.select(dayofweek(df.date)).first()
         self.assertEqual(row[0], 2)
 
+    # Test added for SPARK-37738; change Python API to accept both col & int as input
+    def test_date_add_function(self):
+        dt = datetime.date(2021, 12, 27)
+        df = self.spark.createDataFrame([Row(date=dt, add=2)])
+        # default number in Python gets converted to LongType column
+        df = df.withColumn("add", col("add").cast("integer"))
+
+        row_via_col_addition = df.select(date_add(df.date, df.add)).first()
+        self.assertEqual(row_via_col_addition[0], datetime.date(2021, 12, 29))
+        row_via_scalar_addition = df.select(date_add(df.date, 3)).first()
+        self.assertEqual(row_via_scalar_addition[0], datetime.date(2021, 12, 30))
```

Comment:
We could skip multiple jobs here (and add tests for column name as well):

```
self.assertEqual(all(df.select(
    date_add(df.date, df.add) == datetime.date(2021, 12, 29),
    date_add(df.date, "add") == datetime.date(2021, 12, 29),
    date_add(df.date, 3) ==  datetime.date(2021, 12, 30)
).first()))
```

It is not much, but we have a lot of functions and these quickly add up.

Reply:
Completely agree, and definitely much easier to read